cmd/nerdctl/run.go

```diff
@@ -660,6 +673,46 @@ func generateRootfsOpts(ctx context.Context, client *containerd.Client, platform
 		opts = append(opts, oci.WithProcessArgs(processArgs...))
 	}
 
+	initProcessFlag, err := cmd.Flags().GetBool("init")
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	initBinaryPath, err := cmd.Flags().GetString("init-path")
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	if initProcessFlag {
+		var binaryPath string
+		binaryPath, err := exec.LookPath(nerdctlInitBinary)
+		if err != nil && !errors.Is(err, exec.ErrNotFound) {
+			return nil, nil, nil, err
+		}
+		if err != nil && errors.Is(err, exec.ErrNotFound) {
+			binaryPath, err = exec.LookPath(tiniInitBinary)
+			if err != nil && !errors.Is(err, exec.ErrNotFound) {
+				return nil, nil, nil, err
+			}
+			if err != nil && errors.Is(err, exec.ErrNotFound) {
+				if initBinaryPath != "" {
+					binaryPath = initBinaryPath
+				} else {
+					return nil, nil, nil, fmt.Errorf(`init binary not found, if you want to use --init,please make use a valided binary named nerdctl-init or tini is existed in the PATH or please use the --init-path to specify the path`)
+				}
+			}
+		}
+
+		opts = append(opts, func(_ context.Context, _ oci.Client, _ *containers.Container, spec *oci.Spec) error {
+			spec.Process.Args = append([]string{nerdctlInitBinaryPath, "--"}, spec.Process.Args...)
+			spec.Mounts = append([]specs.Mount{{
+				Destination: nerdctlInitBinaryPath,
```

Comment:
This path should not be hard-coded as `/sbin/nerdctl-init`, it confuses users when debug.

Reply:
Just keep in sync with docker...